src/lib/mina_base/zkapp_basic.ml

```diff
@@ -198,8 +198,7 @@ module Set_or_keep = struct
         ~back:(function
           | { Flagged_option.is_some = true; data = x } ->
               Set (Option.value_exn (to_option x))
-          | { Flagged_option.is_some = false; data = x } ->
-              assert (Option.is_none (to_option x)) ;
```

Comment:
The invariant that it's supposed to check doesn't hold because we're doing some funky magic in the `Typ.t` that wraps it.. Couldn't find a clean way to work around it, so I just removed it.